src/ram-lib-mapping.ts

```diff
@@ -0,0 +1,532 @@
+  updateProjectIfMinterConfigIsActive(
+    minterProjectAndConfig.project,
+    projectMinterConfig,
+    event.block.timestamp
+  );
+}
+
+export function handleNumTokensInAuctionUpdated(
+  event: NumTokensInAuctionUpdated
+): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // update num tokens in auction detail in project minter configuration extraMinterDetails json field
+  let auctionConfigUpdateDetails: TypedMap<
+    string,
+    JSONValue
+  > = createTypedMapFromEntries([
+    {
+      key: "auctionNumTokensInAuction",
+      value: toJSONValue(event.params.numTokensInAuction.toString())
+    }
+  ]);
+
+  mergeProjectMinterConfigExtraMinterDetails(
+    projectMinterConfig,
+    auctionConfigUpdateDetails
+  );
+
+  projectMinterConfig.save();
+
+  // induce sync if the project minter configuration is the active one
+  updateProjectIfMinterConfigIsActive(
+    minterProjectAndConfig.project,
+    projectMinterConfig,
+    event.block.timestamp
+  );
+}
+
+export function handleBidCreated(event: BidCreated): void {
+  const minterProjectAndConfig = loadOrCreateMinterProjectAndConfigIfProject(
+    event.address, // minter
+    event.params.coreContract,
+    event.params.projectId,
+    event.block.timestamp
+  );
+  if (!minterProjectAndConfig) {
+    // project wasn't found, warning already logged in helper function
+    return;
+  }
+
+  const projectMinterConfig = minterProjectAndConfig.projectMinterConfiguration;
+  // Get the auction base price
+  const currentProjectMinterConfigDetails = getProjectMinterConfigExtraMinterDetailsTypedMap(
+    projectMinterConfig
+  );
+  const auctionBasePrice = currentProjectMinterConfigDetails.get(
+    "auctionBasePrice"
+  );
+
+  if (!auctionBasePrice) {
+    log.warning(
+      "Auction base price for minter {} not found for core contract {}",
+      [event.address.toString(), event.params.coreContract.toHexString()]
+    );
+    return;
+  }
+  const bidValue = slotIndexToBidValue(
+    auctionBasePrice.toString(),
+    event.params.slotIndex
+  );
+
+  const fullProjectId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.projectId
+  );
+
+  const ramBidId = generateRAMMinterBidId(
+    event.address.toHexString(),
+    fullProjectId,
+    event.params.bidId.toString()
+  );
+  // @dev: a bid with this ID should not already exist for the RAM Minter
+  const bid = new Bid(ramBidId);
+  if (!bid) {
+    log.warning("Bid ID {} not found for minter {}", [
+      ramBidId,
+      event.address.toHexString()
+    ]);
+    return;
+  }
```

Comment:
I'm a bit confused by this. `new Bid` will always create a new Bid entity so the `!bid` check below will never be true. Based on the comment I think this should be something like 
```suggestion
  let bid = Bid.load(ramBidId);
  if (bid) {
    log.warning("Bid ID {} already exists for minter {}", [
      ramBidId,
      event.address.toHexString()
    ]);
    return;
  }
  bid = new Bid(ramBidId);
```

Reply:
Good call, this makes more sense